### PR TITLE
Fix date bug

### PR DESCRIPTION
**Pull request by gbo-redspher**

### CODE REVIEW

File: structures/case-service/error.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+const isUuid = require('is-uuid')
+const { fault } = require('./fault')
+const { audit } = require('./audit')
+const { zouloudate } = require('../lib')
+const uuidv4 = require('uuid').v4
+
+const Uuid = s.define('Uuid', (value) => isUuid.v4(value))
+const ZoulouDate = zouloudate(s)
+
+const error = s.object({
```
Comment (fanshan) [inline structures/case-service/error.js:11]: `Case` here

File: structures/case-service/error.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  id: s.defaulted(s.optional(Uuid), uuidv4()),
+  timestamp: s.defaulted(ZoulouDate, () => (new Date()).toISOString()),
+  message: s.size(s.string(), 2, 254),
+  owner: s.optional(s.size(s.string(), 2, 32)),
+  error_code: s.integer(),
```
Comment (fanshan) [inline structures/case-service/error.js:16]: Should be string

File: structures/case-service/errorList.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+const s = require('superstruct')
+const { error } = require('./error')
+
+const errorList = s.object({
```
Comment (fanshan) [inline structures/case-service/errorList.js:4]: CaseList

File: structures/case-service/errorList.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+  errors: s.array(error),
```
Comment (fanshan) [inline structures/case-service/errorList.js:5]: cases

File: structures/case-service/errorList.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+const s = require('superstruct')
```
Comment (fanshan) [inline structures/case-service/errorList.js:1]: No need to validate the output of the case service

**Review by fanshan — DISMISSED** [review]
Case structure should be present directly on the Case Service not on rtech-struct

File: structures/lib.js [hunk 6]
```diff
@@ -4,6 +4,20 @@ const zdReg = /^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(\.\d{3})?Z$/
...
 
+exports.ExistingDate = s.define('ExistingDate', (date) => {
+  try {
+    const newIsoDate = new Date(date).toISOString()
+    const dateZReplaced = date.replace('Z', '.000Z')
+
+    if (date === newIsoDate || dateZReplaced === newIsoDate) {
+      return true
+    }
+    return false
```
Comment (glickel) [inline structures/lib.js:15]:
```suggestion
    new Date(date).toISOString()
    return zdReg.test(date)
```

**Review by glickel — CHANGES_REQUESTED** [review]

**Comment by gbo-redspher** [comment]
@glickel @fanshan What should we do with this one ?

**Pull request merged**